homeassistant/components/sensor/min_max.py

```diff
@@ -98,7 +100,7 @@ def async_min_max_sensor_state_listener(entity, old_state, new_state):
             if self._unit_of_measurement != new_state.attributes.get(
                     ATTR_UNIT_OF_MEASUREMENT):
                 _LOGGER.warning("Units of measurement do not match")
-                return
```

Comment:
I had a look into persistent notification and came to the conclusion that it's a bit overkill.
I just re-implemented handling of mismatch in units and set value to unknown plus unit to ERR. I have re-added your test case for this,